pkg/stream/consumer_base.go

```diff
@@ -115,11 +116,20 @@ func (c *baseConsumer) run(kernelCtx context.Context, inputRunner func(ctx conte
 	defer c.logger.Info("leaving consumer %s", c.name)
 	c.logger.Info("running consumer %s with input %s", c.name, c.settings.Input)
 
+	dyingBackgroundCtxWithContainer, err := appctx.CopyContainer(kernelCtx, context.Background())
+	if err != nil {
+		return err
+	}
+
 	// create ctx whose done channel is closed on dying coffin
-	cfn, dyingCtx := coffin.WithContext(context.Background())
+	cfn, dyingCtx := coffin.WithContext(dyingBackgroundCtxWithContainer)
 
+	manualBackgroundCtxWithContainer, err := appctx.CopyContainer(kernelCtx, context.Background())
```

Comment:
You only need a single base context here - a value context on top of a background context will never close their done channel, only the later wrapping (using `coffin.WithContext` and `cfn.Context`) allows them to die